adapters/jwplayer/utils_test.go

```diff
@@ -0,0 +1,239 @@
+		Description: description,
+	})
+	content := openrtb2.Content{
+		URL:   "//test.medial.url/file.mp4",
+		Title: "Test title",
+		Ext:   descriptionExt,
+	}
+
+	metadata := ParseContentMetadata(content)
+	assert.Equal(t, content.URL, metadata.Url)
+	assert.Equal(t, content.Title, metadata.Title)
+	assert.Equal(t, description, metadata.Description)
+}
+
+func TestGetExistingJwpsegs(t *testing.T) {
+	externalSegments1 := []openrtb2.Segment{{Value: "sthg"}, {Value: "else"}}
+	externalData1 := openrtb2.Data{Name: "external", Segment: externalSegments1}
+
+	externalSegments2 := []openrtb2.Segment{{Value: "sthg2"}, {Value: "else2"}}
+	externalData2 := openrtb2.Data{Name: "external number 2", Segment: externalSegments2}
+
+	jwSegments := []openrtb2.Segment{{Value: "1"}, {Value: "2"}}
+	dataExt := DataExt{Segtax: 502}
+	ext, _ := json.Marshal(dataExt)
+	jwData := openrtb2.Data{Name: "jwplayer.com", Segment: jwSegments, Ext: ext}
+
+	externalSegments3 := []openrtb2.Segment{{Value: "3"}, {Value: "4"}}
+	dataWithoutSegtax := openrtb2.Data{Name: "jwplayer.com", Segment: externalSegments3}
+
+	externalSegments4 := []openrtb2.Segment{{Value: "5"}, {Value: "6"}}
+	dataWithWrongName := openrtb2.Data{Name: "other.com", Segment: externalSegments4, Ext: ext}
+
+	dataWithEmptySegments := openrtb2.Data{Name: "jwplayer.com", Ext: ext}
+
+	jwpsegs := GetExistingJwpsegs([]openrtb2.Data{externalData1, dataWithoutSegtax, dataWithWrongName, dataWithEmptySegments, jwData, externalData2})
+	expectedJwpsegs := []string{"1", "2"}
+
+	assert.Len(t, jwpsegs, len(expectedJwpsegs))
+	assert.ElementsMatch(t, jwpsegs, expectedJwpsegs)
+}
+
+func TestHasJwpsegs(t *testing.T) {
+	segments := []openrtb2.Segment{{
+		Value: "88888888",
+	}}
+	jwDatumExt, _ := json.Marshal(DataExt{Segtax: jwplayerSegtax})
+	datum := openrtb2.Data{
+		Name:    "jwplayer.com",
+		Segment: segments,
+		Ext:     jwDatumExt,
+	}
+
+	assert.True(t, HasJwpsegs(datum))
+
+	datum.Name = "other"
+	assert.False(t, HasJwpsegs(datum))
+
+	datum.Name = "jwplayer.com"
+	datum.Ext, _ = json.Marshal(ContentExt{Description: "descr"})
+	assert.False(t, HasJwpsegs(datum))
+
+	datum.Ext = jwDatumExt
+	datum.Segment = []openrtb2.Segment{}
+	assert.False(t, HasJwpsegs(datum))
+}
+
+func TestIsValidMediaUrl(t *testing.T) {
+	assert.False(t, IsValidMediaUrl(""))
+	assert.False(t, IsValidMediaUrl("nothing"))
+	assert.False(t, IsValidMediaUrl("media.mp4"))
+	assert.False(t, IsValidMediaUrl("file://hello.com/media.mp4"))
+	assert.False(t, IsValidMediaUrl("localhost:9999/hello.com/media.mp4"))
+	assert.False(t, IsValidMediaUrl("0.0.0.0:9999/hello.com/media.mp4"))
+	assert.False(t, IsValidMediaUrl("/hello.com/media.mp4"))
+
+	assert.True(t, IsValidMediaUrl("//hello.com/media.mp4"))
+	assert.True(t, IsValidMediaUrl("http://hello.com/media.mp4"))
+	assert.True(t, IsValidMediaUrl("https://hello.com/media.mp4"))
+	assert.True(t, IsValidMediaUrl("https://hello.com/media.mp4?additional=sthg"))
```

Comment:
@pgodzin does this cover all media url scenarios ?

Reply:
the full spec is rather large and complicated, so it looks like it covers the main cases, but there's an rfc for full coverage: https://github.com/dgerber/rfc3987/blob/master/rfc3987.py

idk the js world very well, but wonder if it makes sense to use a 3p library like https://www.npmjs.com/package/valid-url

Reply:
also something like `www.example.com/video.mp4` should fail with no scheme

Reply:
this also seems like a way to do it https://stackoverflow.com/a/43467144/1530254

Reply:
why in js ? This is in go for Prebid Server

Reply:
I added a case for `www.example.com/video.mp4`

Reply:
ah sorry right, forgot this was go -- I see you're using `url.ParseRequestURI` which should cover most of the cases

are the other 3 conditions (!isLocalFile && !isLocalHost && !isRelativePath) not caught by ParseRequestURI?

Reply:
that's correct, `ParseRequestURI` is not as good as we hoped